src/App.tsx

```diff
@@ -1,40 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [visibleFilms, setvisibleFilms] = useState([...moviesFromServer]);
+  const thefilter = (keyWord: string) => {
+    setvisibleFilms([...moviesFromServer].filter((movie) => {
+      const { title, description } = movie;
 
-export class App extends React.Component<{}, State> {
-  state: State = {};
+      return title.toLocaleLowerCase().includes(keyWord.toLowerCase())
+        || description.toLowerCase().includes(keyWord.toLowerCase());
+    }));
+  };
```

Comment:
Don't save filtered movies in state, create function preparedFilms and save result in const. Can use `useMemo` here